01_explore_data/nbconverted/missing_genes.py

```diff
@@ -0,0 +1,268 @@
+    valid_genes_datasets[dataset] = valid_genes
+    print('{}:'.format(dataset),
+          '{} valid /'.format(np.count_nonzero(valid_genes)),
+          '{} total'.format(len(valid_genes)))
+
+
+# ### Compare valid genes to previously profiled gene set
+
+# In[5]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+from adjustText import adjust_text
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+
+
+# In[6]:
+
+
+dataset_directory = {
+    'expression': 'vogelstein_expression_only',
+    'methylation': 'compressed_results',
+    'all': 'all_data_types_results'
+}
+dataset = 'expression'
+
+# set results directory
+results_dir = Path(cfg.results_dirs['mutation'],
+                   dataset_directory[dataset],
+                   'gene').resolve()
+# set significance cutoff after FDR correction
+SIG_ALPHA = 0.001
+
+# load raw data
+results_df = au.load_stratified_prediction_results(results_dir, 'gene')
+results_df = results_df[results_df.training_data == 'expression']
+
+# make sure that we're correctly pointing to raw data for non-methylation data types
+# and that we have data for two replicates (two random seeds)
+print(results_df.shape)
+print(results_df.seed.unique())
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# In[7]:
+
+
+all_results_df = au.compare_results(results_df,
+                                    identifier='identifier',
+                                    metric='aupr',
+                                    correction=True,
+                                    correction_method='fdr_bh',
+                                    correction_alpha=SIG_ALPHA,
+                                    verbose=True)
+all_results_df.rename(columns={'identifier': 'gene'}, inplace=True)
+all_results_df.sort_values(by='p_value').head(10)
+
+
+# In[8]:
+
+
+prev_genes = all_results_df.gene.unique()
+cur_genes = valid_genes_datasets[dataset][
+    valid_genes_datasets[dataset] == True
+].index.values
+prev_not_cur = [g for g in prev_genes if g not in cur_genes]
+cur_not_prev = [g for g in cur_genes if g not in prev_genes]
+print(prev_not_cur)
+print(cur_not_prev)
+
+
+# In[9]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.set_style('whitegrid')
+
+all_results_df['nlog10_p'] = -np.log10(all_results_df.corr_pval)
+
+# all plots should have the same axes for a fair comparison
+xlim = (-0.2, 1.0)
+y_max = all_results_df.nlog10_p.max()
+ylim = (0, y_max+3)
+
+# function to add gene labels to points
+def label_points(x, y, gene, ax):
+    text_labels = []
+    a = pd.DataFrame({'x': x, 'y': y, 'gene': gene})
+    for i, point in a.iterrows():
+        if point['y'] > -np.log10(SIG_ALPHA / 1000):
+            text_labels.append(
+                ax.text(point['x'], point['y'], str(point['gene']))
+            )
+    return text_labels
+
+# plot mutation prediction from expression, in a volcano-like plot
```

Comment:
Is there a reason the vanishing genes are all at the bottom of the distribution instead of uniformly distributed? Is it because genes with bigger effect sizes are better studied/show up in more data types?

Reply:
The reason they were "vanishing" before is that there was only a few positive labels, so sometimes I'd (randomly) end up with a CV split with no positive labels, and AUROC/AUPR are undefined in that case. But I think because there were so few positive labels in these cases, we weren't able to train good classifiers anyway, so they were at the bottom because the label imbalance was so extreme.

So I do think what you're saying is the right idea, although I'd put it like this: the genes with large effect sizes in these experiments tend to have more positive labels/be more frequently mutated, so I'm not surprised that we're not losing any of them with this change.

It's an interesting thought, though - I'm sure there are some genes that are infrequently mutated and have a large effect size (can't think of an example off the top of my head), but I'm not sure our current classification approach would be effective at identifying those genes. Maybe an opportunity for better methods in the future!